test/Search_UnitTestHelpers.hpp

```diff
@@ -144,58 +176,67 @@ zip(ExecutionSpace const &space, Kokkos::View<int *, DeviceType> indices,
 }
 
 #ifdef ARBORX_ENABLE_MPI
-template <typename Tree, typename Queries>
-auto query_with_distance(Tree const &tree, Queries const &queries,
+template <typename ExecutionSpace, typename Tree, typename Queries>
+auto query_with_distance(ExecutionSpace const &exec_space, Tree const &tree,
+                         Queries const &queries,
                          std::enable_if_t<is_distributed<Tree>{}> * = nullptr)
 {
-  using device_type = typename Tree::device_type;
-  Kokkos::View<int *, device_type> offsets("Testing::offsets", 0);
-  Kokkos::View<int *, device_type> indices("Testing::indices", 0);
-  Kokkos::View<int *, device_type> ranks("Testing::ranks", 0);
-  Kokkos::View<float *, device_type> distances("Testing::distances", 0);
-  using ExecutionSpace = typename device_type::execution_space;
-  ExecutionSpace space;
-  ArborX::Details::DistributedTreeImpl<device_type>::queryDispatchImpl(
-      ArborX::Details::NearestPredicateTag{}, tree, space, queries, indices,
-      offsets, ranks, &distances);
+  using MemorySpace = typename Tree::memory_space;
+  using DeviceType = Kokkos::Device<ExecutionSpace, MemorySpace>;
+
+  Kokkos::View<int *, MemorySpace> offsets("Testing::offsets", 0);
+  Kokkos::View<int *, MemorySpace> indices("Testing::indices", 0);
+  Kokkos::View<int *, MemorySpace> ranks("Testing::ranks", 0);
+  Kokkos::View<float *, MemorySpace> distances("Testing::distances", 0);
+  ArborX::Details::DistributedTreeImpl<DeviceType>::queryDispatchImpl(
+      ArborX::Details::NearestPredicateTag{}, tree, exec_space, queries,
+      indices, offsets, ranks, &distances);
 
-  auto values = zip(space, indices, ranks, distances);
+  auto values = zip(exec_space, indices, ranks, distances);
 
   return make_compressed_storage(
       Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offsets),
       Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, values));
 }
 #endif
 
-#define ARBORX_TEST_QUERY_TREE_WITH_DISTANCE(tree, queries, reference)         \
-  BOOST_TEST(query_with_distance(tree, queries) == (reference),                \
+#define ARBORX_TEST_QUERY_TREE_WITH_DISTANCE(exec_space, tree, queries,        \
+                                             reference)                        \
+  BOOST_TEST(query_with_distance(exec_space, tree, queries) == (reference),    \
              boost::test_tools::per_element());
 
-template <typename Tree>
-auto make(std::vector<ArborX::Box> const &b)
+template <typename DeviceType>
+ArborX::BVH<typename DeviceType::memory_space>
+makeTree(std::vector<ArborX::Box> const &b)
```

Comment:
That's not a good idea.  You want to template on the tree or you won't be able to use with other data structures.
```suggestion
template <typename Tree, typename ExecutionSpace>
Tree make(ExecutionSpace const &exec_space, std::vector<ArborX::Box> const &b)
```

Reply:
That's what I did originally. I switched it to this version as a) after looking around, it's only used to make BVH; and b) it allowed me to not do an extra `ExecutionSpace{}` argument everywhere. I'd prefer to leave it like this, and change it if the need arises (which may be soon, but now immediately).

Reply:
Please fix in this PR

Reply:
You can do `ExecutionSpace exec_space{};` right after the `using` declaration of the space.  Make sure you use it in the tree construction.

Reply:
Fixed.